payjoin-cli/src/app/v2/mod.rs

```diff
@@ -157,15 +158,12 @@ impl AppTrait for App {
             unwrap_ohttp_keys_or_else_fetch(&self.config, None, self.relay_manager.clone())
                 .await?
                 .ohttp_keys;
+        let receiver_builder =
+            ReceiverBuilder::new(address, self.config.v2()?.pj_directory.clone(), ohttp_keys)?
+                .with_amount(amount);
         let persister = ReceiverPersister::new(self.db.clone())?;
-        let session = Receiver::create_session(
-            address,
-            self.config.v2()?.pj_directory.clone(),
-            ohttp_keys,
-            None,
-            Some(amount),
-        )?
-        .save(&persister)?;
+        let session = receiver_builder.build().save(&persister)?;
+
```

Comment:
I think `let session = ReceiverBuilder::new() ..build().save(&persister)?;` All chained is the way to go